### DISCUSSION TITLE
Inverse edge setting of the edge && the unit setting of the property in schema

### DISCUSSION THREAD

**Discussion by bidiudiu** [JanusGraph/janusgraph#2685] [Q&A]
（1） I want to set the inverse edge of an edge so that some missing data can be filled automatically. For example, People-wife->People, People-husband->People. ‘wife’ is ‘husband’‘s inverse edge and 'husband' is 'wife''s inverse edge.
（2）I also want to store unit information in schema. For example, property 'price''s unit is '$' and property 'height''s unit is 'meter'.

However, I did not find the above two settings in the official description of the **schema**. Do you have similar needs and how did you solve them? Thank you

**Comment by vtslab** [comment]
Hi, 
Regarding 1, inference of relations is not supported. It is possible though to write a transaction log with a specific identifier that can be picked up by a separate process that could do the inference, see:
[https://docs.janusgraph.org/advanced-topics/janusgraph-bus/](https://docs.janusgraph.org/advanced-topics/janusgraph-bus/)

Regarding 2, janusgraph supports meta-properties, as described in:
[https://tinkerpop.apache.org/docs/current/reference/#vertex-properties](https://tinkerpop.apache.org/docs/current/reference/#vertex-properties). 
So you could add the unit as a meta-property.
Note however that:
1. they are still undocumented in JanusGraph ([https://github.com/JanusGraph/janusgraph/issues/225](https://github.com/JanusGraph/janusgraph/issues/225))
2. they are controversial in Apache TinkerPop (sorry, no source specified, look for discussions on TinkerPop 4)
So maybe you are better off with a schema definition in your application that you use to create the schema in JanusGraph on the one hand and look up the unit on the other hand.
Best wishes,     Marc

**Reply by bidiudiu** [reply]
I see. Thanks Marc.
